subsys/testsuite/ztest/Kconfig

```diff
@@ -73,6 +79,20 @@ config ZTEST_ASSERT_HOOK
 	  error test case. Remember to add ignore_fault tag in yaml file when
 	  using twister to run testing.
 
+if ZTEST_NEW_API
```

Comment:
Just a drive-by comment; feel free to ignore.

Words like "new" or "modern" have a habit of aging like milk. Have you considered something like "ZTEST_API_V2" instead?

Reply:
Sure, I was using `NEW` because I'm planning to deprecate the other one and eventually removing this Kconfig entirely. Let me know if you'd still like it to be `V2`

Reply:
I also like "new" because it sits as a frequent reminder to complete the migration.